### PR TITLE
Fix cast to array bug

### PR DESCRIPTION
**Pull request by kapersoft**
Fixes https://github.com/spatie/laravel-activitylog/issues/680; test for the issue is included.

Let me know if you have any questions or remarks.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     },
     "require-dev": {
         "ext-json": "*",
+        "orchestra/testbench": "^5.0",
```
Comment (Gummibeer) [inline composer.json]: We still support L6 - so we also need Orchestra 4.

File: src/Traits/DetectsChanges.php [hunk 2]
```diff
@@ -133,8 +133,9 @@ public static function logChanges(Model $model): array
                     static::getModelAttributeJsonValue($model, $attribute)
                 );
             } else {
-                $changes[$attribute] = $model->getAttribute($attribute);
-
+                $changes[$attribute] = $model->hasCast($attribute, 'array') && is_array($model->attributes[$attribute])
```
Comment (Gummibeer) [inline src/Traits/DetectsChanges.php]: The `json` cast is the same. And does this happen with `collections`/`objects` too? 🤔

**Comment by Gummibeer** [comment]
At the end I'm still not sure what we are doing wrong? 🤔 
Isn't this something to get fixed in the core? I mean in theory this should also happen if you call `getAttribute()` your own?

**Comment by ron-florax** [comment]
If this helps, I initially thought it was an issue in the framework, but it doesn't seem to be: https://github.com/laravel/framework/issues/31790

**Comment by canvural** [comment]
We also had this issue in our projects. I did some digging. I think I found something. I'm not sure.

[Here](https://github.com/spatie/laravel-activitylog/blob/master/src/Traits/DetectsChanges.php#L21) it's calling `getOriginal` method. `getOriginal` method returns all the attributes, but also casts them. 
So later when `logChanges` is working, when it calls `$model->getAttribute($attribute);` attribute value is already casted to array, so it throws the error.

I'm not sure this is the cause, but maybe you can investigate more.

**Comment by SudoGetBeer** [comment]
@canvural Interesting.

Maybe it is this: https://laravel.com/docs/7.x/upgrade#factory-types

> The $model->getOriginal() method will now respect any casts defined on the model. Previously, this method returned the uncast, raw attributes. If you would like to continue retrieving the raw, uncast values, you may use the getRawOriginal method instead.

**Comment by canvural** [comment]
Yeah, that looks like the reason.

So, maybe this issue can be solved by using `getRawOriginal`

**Comment by Gummibeer** [comment]
Thanks @kapersoft for your work here! 🎉  I was able to use everything and have extended your PR/branch.
I will close this PR in favor of #688 .

**Comment by Gummibeer** [comment]
released: https://github.com/spatie/laravel-activitylog/releases/tag/3.11.2

**Pull request closed without merging**